### ISSUE TITLE
when i generate onnx file ,i got the error.Please give me some tips.thanks!

### ISSUE THREAD

**Issue by LCWdmlearning** [xingyizhou/CenterNet#72]
model = DLASeg('dla{}'.format(34), {'hm': 80, 'wh': 2, 'reg': 2},
               pretrained=True,
               down_ratio=4,
               final_kernel=1,
               last_level=5,
               head_conv=256)
model = model.to('cuda')
model.eval()
x = torch.randn(1, 3, 512, 512)
x = x.to('cuda')
torch_out = torch.onnx._export(model, x, "model.onnx", export_params=True)

error：
RuntimeError: Attempted to trace DCNv2Function, but tracing of legacy functions is not supported 

I checked Google and it seems that the type of (torch.autograd.Function) that not support to conversion onnx

**Comment by xingyizhou** [comment]
I honestly have no idea about this. Please check your pytorch/ cuda/ python versions match the reference.

**Comment by cwjhx** [comment]
> my code:
> model = DLASeg('dla{}'.format(34), {'hm': 80, 'wh': 2, 'reg': 2},
> pretrained=True,
> down_ratio=4,
> final_kernel=1,
> last_level=5,
> head_conv=256)
> model = model.to('cuda')
> model.eval()
> x = torch.randn(1, 3, 512, 512)
> x = x.to('cuda')
> torch_out = torch.onnx._export(model, x, "model.onnx", export_params=True)
> 
> error：
> RuntimeError: Attempted to trace DCNv2Function, but tracing of legacy functions is not supported
> 
> I checked Google and it seems that the type of (torch.autograd.Function) that not support to conversion onnx

hello ! I meet same problems.Have you solved it?

**Comment by LCWdmlearning** [comment]
> > my code:
> > model = DLASeg('dla{}'.format(34), {'hm': 80, 'wh': 2, 'reg': 2},
> > pretrained=True,
> > down_ratio=4,
> > final_kernel=1,
> > last_level=5,
> > head_conv=256)
> > model = model.to('cuda')
> > model.eval()
> > x = torch.randn(1, 3, 512, 512)
> > x = x.to('cuda')
> > torch_out = torch.onnx._export(model, x, "model.onnx", export_params=True)
> > error：
> > RuntimeError: Attempted to trace DCNv2Function, but tracing of legacy functions is not supported
> > I checked Google and it seems that the type of (torch.autograd.Function) that not support to conversion onnx
> 
> hello ! I meet same problems.Have you solved it?

no,I haven't solved it. some layers that use the Function type(torch.autograd.Function).and hat not support to conversion onnx.

**Comment by LCWdmlearning** [comment]
> > my code:
> > model = DLASeg('dla{}'.format(34), {'hm': 80, 'wh': 2, 'reg': 2},
> > pretrained=True,
> > down_ratio=4,
> > final_kernel=1,
> > last_level=5,
> > head_conv=256)
> > model = model.to('cuda')
> > model.eval()
> > x = torch.randn(1, 3, 512, 512)
> > x = x.to('cuda')
> > torch_out = torch.onnx._export(model, x, "model.onnx", export_params=True)
> > error：
> > RuntimeError: Attempted to trace DCNv2Function, but tracing of legacy functions is not supported
> > I checked Google and it seems that the type of (torch.autograd.Function) that not support to conversion onnx
> 
> hello ! I meet same problems.Have you solved it?

no,I haven't solved it. some layers that use the Function type(torch.autograd.Function).and hat not support to conversion onnx.

**Comment by cwjhx** [comment]
when I run：torch_out = torch.onnx._export(model, x, "model.onnx", export_params=True)
error：
ValueError: Auto nesting doesn't know how to process an input object of type int. Accepted types: Tensors, or lists/tuples of them

**Comment by LCWdmlearning** [comment]
> when I run：torch_out = torch.onnx._export(model, x, "model.onnx", export_params=True)
> error：
> ValueError: Auto nesting doesn't know how to process an input object of type int. Accepted types: Tensors, or lists/tuples of them
i don'not know how to solved it. can you skip the first problems?

**Comment by cwjhx** [comment]
my code：
checkpoint = torch.load(model_path)
model = DLASeg(...）
model.load_state_dict(checkpoint['state_dict'])
....
torch_out = torch.onnx._export(model, x, "model.onnx", export_params=True)

**Comment by HiminDong** [comment]
it may be the torch version issue

**Comment by leonzgtee** [comment]
> > when I run：torch_out = torch.onnx._export(model, x, "model.onnx", export_params=True)
> > error：
> > ValueError: Auto nesting doesn't know how to process an input object of type int. Accepted types: Tensors, or lists/tuples of them
> > i don't know how to solved it. can you skip the first problems?

as showed, you need to input list when transform to onnx, i.e. you need to modify the format of ret in network to list

**Comment by caixhstrive** [comment]
this repository maybe help, https://github.com/CaoWGG/TensorRT-CenterNet/blob/master/readme/ctdet2onnx.md
